node/src/BaseClient.ts

```diff
@@ -318,28 +318,80 @@ export type SortedSetDataType = {
 }[];
 
 /**
- * This function converts an input from HashDataType or Record types to HashDataType.
- *
- * @param fieldsAndValues - field names and their values.
- * @returns HashDataType array containing field names and their values.
+ * Data type which represents how data are returned from hashes or insterted there.
+ * Similar to `Record<GlideString, GlideString>` - see {@link GlideRecord}.
  */
-export function convertFieldsAndValuesForHset(
-    fieldsAndValues: HashDataType | Record<string, GlideString>,
-): HashDataType {
-    let finalFieldAndValues = [];
-
-    if (!Array.isArray(fieldsAndValues)) {
-        finalFieldAndValues = Object.entries(fieldsAndValues).map((e) => {
-            return { field: e[0], value: e[1] };
-        });
-    } else {
-        finalFieldAndValues = fieldsAndValues;
+export type HashDataType = {
+    /** The hash element name. */
+    field: GlideString;
+    /** The hash element value. */
+    value: GlideString;
+}[];
+
+/**
+ * Data type which reflects now stream entries are returned.
+ * The keys of the record are stream entry IDs, which are mapped to key-value pairs of the data.
+ */
+export type StreamEntryDataType = Record<string, [GlideString, GlideString][]>;
+
+/**
+ * @internal
+ * Convert `GlideRecord<number>` recevied after resolving the value pointer into `SortedSetDataType`.
+ */
+function convertGlideRecordForZSet(
+    res: GlideRecord<number>,
+): SortedSetDataType {
+    return res.map((e) => {
+        return { element: e.key, score: e.value };
+    });
+}
+
+/**
+ * @internal
+ * Downcast `GlideRecord` to `Record`. Use if you are 146% aware that `data` keys are always strings.
+ */
+export function glideRecordToRecord<T>(
```

Comment:
There is a comment about renaming this in https://github.com/Bit-Quill/valkey-glide/pull/421/files
Comment: Rename this function to castGlideRecordToRecord